recipes/protobuf/all/conanfile.py

```diff
@@ -48,15 +48,16 @@ def configure(self):
         if self.options.shared:
             del self.options.fPIC
 
-            compiler_version = Version(self.settings.compiler.version.value)
-            if compiler_version < "14":
-                raise ConanInvalidConfiguration("On Windows Protobuf can only be built with "
-                                                "Visual Studio 2015 or higher.")
-            if self.options.shared and "MT" in self.settings.compiler.runtime:
-                raise ConanInvalidConfiguration("Protobuf can't be build with shared + MT(d) runtimes")
-
-        if tools.is_apple_os(self.settings.os) and self.options.shared:
-            raise ConanInvalidConfiguration("Protobuf could not be build as shared library for Mac.")
+            if self.settings.os == "Windows":
+                compiler_version = Version(self.settings.compiler.version.value)
+                if compiler_version < "14":
+                    raise ConanInvalidConfiguration("On Windows Protobuf can only be built with "
+                                                    "Visual Studio 2015 or higher.")
+                if self.options.shared and "MT" in self.settings.compiler.runtime:
+                    raise ConanInvalidConfiguration("Protobuf can't be build with shared + MT(d) runtimes")
+
+            if tools.is_apple_os(self.settings.os):
+                raise ConanInvalidConfiguration("Protobuf could not be build as shared library for Mac.")
```

Comment:
```suggestion
        if self.settings.compiler == "Visual Studio":
            if Version(self.settings.compiler.version) < "14":
                raise ConanInvalidConfiguration("On Windows Protobuf can only be built with "
                                                "Visual Studio 2015 or higher.")
            if self.options.shared and "MT" in self.settings.compiler.runtime:
                raise ConanInvalidConfiguration("Protobuf can't be build with shared + MT(d) runtimes")

        if tools.is_apple_os(self.settings.os) and self.options.shared:
            raise ConanInvalidConfiguration("Protobuf could not be build as shared library for Mac.")
```

Reply:
this all block shouldn't be under the condition `self.options.shared` deleting fPIC option.

Reply:
The VS >= 2015 should indeed not be in it, true

Not sure about the runtime. After all Clang for Windows supports those runtimes too, right?

Reply:
But not MinGW